docs(analytics): addingwell support (#903)
front-commerce/developers.front-commerce.com: versioned_docs/version-2.x/advanced/analytics/server-side-tagging-tracking.mdx

---
sidebar_position: 4
title: Server-side tracking / tagging
description:
  This page will guide you trough the setup of Server-side tracking and tagging

>   services

Reviewer: **issue:** I think having a dedicated page isn't relevant. Addingwell is a service like others, and as far as FC is concerned it still is client-side tracking.

**suggestion:** move the How-To under a dedicated subsection of the existing ["Common plugins"](https://deploy-preview-903--heuristic-almeida-1a1f35.netlify.app/docs/2.x/advanced/analytics/plugins#common-plugins) section.

---

<p>{frontMatter.description}</p>

## Addingwell

> ### Requirements
> 
> To use Addingwell, you must have create an
> [Addingwell account](https://www.addingwell.com/)
> 
> ### Setup Addingwell server
> 
> You must follow the official
> [Addingwell Get Started](https://docs.addingwell.com/b/2CA8BCF1-E67D-46DF-A2BE-F93017D8FD7A/Getting-started-with-Addingwell)
> documentation.

Reviewer: **nitpick:** one sentence sections takes too much visual space IMO

**suggestion:** group into a single section

```suggestion
### Prerequisites

Before configuring Addingwell in Front-Commerce, ensure that you have:
- created an [Addingwell account](https://www.addingwell.com/)
- setup your own server, as documented in the [Addingwell Get Started](https://docs.addingwell.com/b/2CA8BCF1-E67D-46DF-A2BE-F93017D8FD7A/Getting-started-with-Addingwell) official guide
```

### With GTM

This guide will show you how to integrate Addingwell with Google Tag Manager.
This way, the Analytics tag will be implemented with Google Tag Manager but the

> traffic will head to your custom server

Reviewer: ```suggestion
traffic will head to your custom server.
```

#### Setup GTM

You must follow the
[Addingwell Send data documentation (Option 1)](https://docs.addingwell.com/b/C45DD221-2773-4DE0-B588-39B60785291A/Send-the-data-to-your-tagging-server)

> #### Setup GTM Analytics plugin in Front-Commerce

Reviewer: **nitpick:** since it is similar to what is already documented, I'd link to the existing section to prevent duplication

**suggestion:** reword into something more concise, such as

```suggestion
#### Configure Google Tag Manager in Front-Commerce

Follow the [Google Tag Manager](/docs/2.x/advanced/analytics/plugins#google-tag-manager) installation procedure documented above.

To use it with Addingwell, you'll have one more step to do: configure …………
```

Then you'll need to update your CSP to allow your Front-Commerce application to
communicate with external services. Let's say that your analytics domain is
`metrics.my-commerce.net`

```js title="src/config/website.js"
module.exports = {
  //....
  contentSecurityPolicy: {
    directives: {
      // highlight-next-line
      scriptSrc: ["metrics.my-commerce.net", "www.googletagmanager.com"], // We need to add Google Tag Manager to allow GTM to inject GA4 tag into our pages
      frameSrc: [],
      styleSrc: [],
      imgSrc: [],
      fontSrc: [],
      // highlight-next-line
      connectSrc: ["metrics.my-commerce.net"],
      baseUri: [],
      childSrc: [],
      defaultSrc: [],
      manifestSrc: [],
      mediaSrc: [],
      objectSrc: [],
      workerSrc: [],
    },
  },
  //....
};
```

### With GA4

#### Setup your server container

You first need to follow the Addingwell
[gtag.js configuration guide (option 3)](https://docs.addingwell.com/b/C45DD221-2773-4DE0-B588-39B60785291A/Send-the-data-to-your-tagging-server).

Please note your custome tagging domain we will need this for the next step (in
the guide, we will consider that the domain is `metrics.my-commerce.net`).

> #### Setup GA4 plugin in Front-Commerce

Reviewer: **nitpick:** similar to the GTM setup, I think mentioning the previous section and an updated config with only the differences highlighted would help keeping it short here
